src/components/uiOptions/js/UIOptions.js

```diff
@@ -53,4 +53,228 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
+                                        args: [
+                                            "{fluid.uiOptions.prefsEditor.localized}.prefsEditorLoader.messageLoader.resources",
+                                            "{fluid.uiOptions.prefsEditor.localized}.model.locale",
+                                            "{fluid.uiOptions.prefsEditor.localized}.prefsEditorLoader.prefsEditor"
+                                        ],
+                                        priority: "before:rerenderPanels",
+                                        namespace: "updateUioPanelLocales"
+                                    },
+                                    {
+                                        funcName: "fluid.prefs.separatedPanel.updateView",
+                                        args: ["{that}"],
+                                        namespace: "rerenderPanels"
+                                    },
+                                    {
+                                        func: "{fluid.uiOptions.prefsEditor.localized}.events.onUioPanelsUpdated",
+                                        namespace: "panelsUpdated",
+                                        priority: "last"
+                                    }]
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    // Extension of lazyLoad grade to add listener for locale change
+    fluid.defaults("fluid.prefs.separatedPanel.localized.lazyLoad", {
+        gradeNames: ["fluid.prefs.separatedPanel.lazyLoad"],
+        listeners: {
+            "{fluid.uiOptions.prefsEditor.localized}.events.onInterfaceLocaleChangeRequested": {
+                func: "{fluid.uiOptions.prefsEditor.localized}.reloadUioMessages",
+                args: ["{arguments}.0.data", "{that}.events.onLazyLoad"],
+                priority: "after:changeLocale",
+                namespace: "reloadUioMessages"
+            }
+        }
+    });
+
+    /* Reloads UIO message bundles with the given locale,
+     * subject to the fallback rules of the resourceLoader and,
+     * if provided, fires an event to trigger lazy loading
+     * - "locale": the locale for which to load messages
+     * - "uioMessageLoaderComponent": the UIO messageLoader component
+     * - "uioMessageLoaderLocalePath": the path on the messageLoader for its locale
+     * - "lazyLoadEvent": the event to fire to trigger lazy loading
+     */
+    fluid.uiOptions.prefsEditor.localized.reloadUioMessages = function (locale, uioMessageLoaderComponent, uioMessageLoaderLocalePath, lazyLoadEvent) {
+        // Set the locale in the resource loader
+        fluid.set(uioMessageLoaderComponent, uioMessageLoaderLocalePath, locale);
+
+        // Force the resource loader to get the new resources
+        fluid.resourceLoader.loadResources(uioMessageLoaderComponent, uioMessageLoaderComponent.resolveResources());
```

Comment:
Push this into the ResourceLoader impl itself rather than calling on a global function